tiledb/cf/netcdf_engine/_array_converters.py

```diff
@@ -190,44 +236,32 @@ def get_query_coordinates(
                 return coordinates for a dense write.
             assigned_dim_values: Values for any non-NetCDF dimensions.
         """
-        query_coords = []
-        for dim_creator in self:
-            if isinstance(dim_creator.base, NetCDF4ToDimConverter):
-                query_coords.append(
-                    dim_creator.base.get_values(netcdf_group, sparse=sparse)
-                )
-            else:
-                if (
-                    assigned_dim_values is None
-                    or dim_creator.name not in assigned_dim_values
-                ):
-                    raise KeyError(f"Missing value for dimension '{dim_creator.name}'.")
-                query_coords.append(assigned_dim_values[dim_creator.name])
+        query_coords = tuple(
+            dim_creator.get_query_coordinates(
+                netcdf_group, sparse, index_slice, assigned_dim_values
+            )
+            for index_slice, dim_creator in zip(indexer, self)
```

Comment:
Does this do the right thing if `indexer` doesn't have the same length as `self`?

Reply:
I'll add an error check.